File: RxHttp/src/main/java/com/github/kongpf8848/rxhttp/request/PostRequest.java
package com.github.kongpf8848.rxhttp.request;

import android.app.Activity;
import android.content.Context;
import android.text.TextUtils;

import androidx.fragment.app.Fragment;

import com.github.kongpf8848.rxhttp.HttpConstants;
import com.github.kongpf8848.rxhttp.util.GsonUtil;

import okhttp3.MediaType;
import okhttp3.RequestBody;

public class PostRequest extends AbsRequest {

    private String type;
    public PostRequest(Context context) {
        super(context);
    }
    public PostRequest(Activity activity) {
        super(activity);
    }
    public PostRequest(Fragment fragment) {
        super(fragment);
    }

    public PostRequest type(String type){
        this.type=type;
        return this;
    }

    @Override
    public RequestBody buildRequestBody() {
        String content= GsonUtil.toJson(getParams());
        if(TextUtils.isEmpty(type)) {
            return RequestBody.create(MediaType.parse(HttpConstants.MIME_TYPE_JSON), content);
        }
        else{
            return RequestBody.create(MediaType.parse(type), content);
        }
    }
}

File: RxHttp/src/main/java/com/github/kongpf8848/rxhttp/request/PostRequest.kt
package com.github.kongpf8848.rxhttp.request

import android.app.Activity
import android.content.Context
import android.text.TextUtils
import androidx.fragment.app.Fragment
import com.github.kongpf8848.rxhttp.HttpConstants
import com.github.kongpf8848.rxhttp.util.GsonUtil
import okhttp3.MediaType.Companion.toMediaTypeOrNull
import okhttp3.RequestBody
import okhttp3.RequestBody.Companion.toRequestBody

open class PostRequest: AbsRequest {

    private var type: String? = null

    constructor(context: Context) : super(context)
    constructor(activity: Activity) : super(activity)
    constructor(fragment: Fragment) : super(fragment)

    fun type(type: String): PostRequest {
        this.type = type
        return this
    }

    public override fun buildRequestBody(): RequestBody? {
        val content = GsonUtil.toJson(getParams())
        return if (TextUtils.isEmpty(type)) {
            content.toRequestBody(HttpConstants.MIME_TYPE_JSON.toMediaTypeOrNull())
        } else {
            content.toRequestBody(type!!.toMediaTypeOrNull())
        }
    }
}